### DISCUSSION TITLE
Why is a QButton taking space when used with a v-if?

### DISCUSSION THREAD

**Discussion by wsw70** [quasarframework/quasar#13336] [General - Components / Directives / etc] [answered]
Consider the following (simplified) code that displays a `q-input` (a `textarea`), and conditionally a button. My expectation was that the  `textarea` would be full width, then shrink when the button is visible, and expand back when the button is off.  What I have is an always-shrinked `textarea`

```
<template>
    <div v-for="scrap in allScraps" :key="scrap.Id"  class="row scrap q-mb-md q-px-sm">
      <q-input v-model="scrap.Text" type="textarea"></q-input>
      <q-btn v-if="showDelete" color="warning" flat icon="delete"></q-btn>
    </div>
<template>

<script lang="ts" setup>
    const showDelete = ref(false)
</script>

<style>
.scrap {
  border-width: 1px;
  border-style: dashed;
  border-color: darkblue;
}
</style>
```

My problem is that `<q-btn>` is taking space even in the `v-if` condition is false. Below is a screenshot for both values:

![image](https://user-images.githubusercontent.com/1345886/166651990-912147eb-bf89-4561-835c-0f9478bd31df.png)

![image](https://user-images.githubusercontent.com/1345886/166652008-94f12488-324c-45da-8c7e-152985ada7a5.png)

**Why isn't the `textarea` expanding to the full width when `v-if` is false?**

When looking,g at DevTools and targeting one of the `div`s (generated in the iteration), I see some kind of margin that is always present:

- when `showDelete` is `false`:

![image](https://user-images.githubusercontent.com/1345886/166653009-65ba6cef-71e3-4002-940e-46a5b604d4e2.png)

- when `showDelete` is `true`:

![image](https://user-images.githubusercontent.com/1345886/166653018-fa336d51-906e-41c6-b369-16abb07baf33.png)

I also tried to change the style to the following, without success (I guess this is because `display: flex;` does not propagate down the DOM:
```

.scrap {
  border-width: 1px;
  border-style: dashed;
  border-color: darkblue;
  display: flex;
}

textarea {
  border: none;
  outline: none;
  flex: 1;
}

```

**Comment by pdanpdan** [comment, marked as answer]
Make a codepen
I don't know what would `flex: 1` do
The button is not rendered, but your text area is not set to flex grow

**Reply by wsw70** [reply]
I will try to make a codepen from my app. 

Re: growth: when the button element is removed from the code, the `textarea` grows to the full width. I was expecting that the `v-if` would do exactly this: to remove the element completely. It would seem that Quasar (?) "provisions" the space no matter the value of `v-if`

Re: `flex: 1;`: https://stackoverflow.com/a/37386525/903011

**Reply by pdanpdan** [reply]
Exactly as expected :) it does something but only the browser knows what
Why don't you use .row on parent and .col on qinput?

And nobody reserves space

**Reply by wsw70** [reply]
Ha! Thanks a lot! I added the `col` class to QInput and it solved the issue! I am not sure why - I had the class `row` applied to the `div` I am iterating on to say "what is in the div should be aligned in a row". Not sure where the `col` comes into play but I will read once again the docs on the grid system. Thanks!

**Comment by pdanpdan** [comment]
Row sets the display flex (direction row) on parent
Col sets flex rules for the children
You can inspect what is applied
